### PR TITLE
Fix #422

### LINKED ISSUE #422: Leaking Private Messages IDOR
Edit the **id** paremeter in a request to **/Messages/GetMessage** to read others chats.

### PR DESCRIPTION
**Pull request by wfinn**
This should fix #422

### CODE REVIEW

File: zapread.com/Controllers/MessagesController.cs [hunk 1]
```diff
@@ -873,18 +873,30 @@ public async Task<JsonResult> SendMessage(int id, string content, bool? isChat)
...
 
         [HttpPost]
-        [AllowAnonymous]
         public JsonResult GetMessage(int id, int userId)
         {
+            var localUserId = User.Identity.GetUserId();
+
+            if(localUserId == null)
+            {
+                return Json(new { success = false, result = "Failure", message = "Error receiving message." });
```
Comment (Horndev) [inline zapread.com/Controllers/MessagesController.cs]: 1.  Should change HTTP response code and have more informative message before returning.
`Response.StatusCode = (int)HttpStatusCode.Unauthorized;`
1.  Should have a descriptive message. `Error verifying logged in user.` or something?

File: zapread.com/Controllers/MessagesController.cs [hunk 2]
```diff
@@ -873,18 +873,30 @@ public async Task<JsonResult> SendMessage(int id, string content, bool? isChat)
...
             {
                 var msg = db.Messages
                     .Include("From")
                     .Include("To")
+                    .Where(m => m.From.AppId == localUserId || m.To.AppId == localUserId)
                     .SingleOrDefault(m => m.Id == id);
 
+                if (msg == null)
+                {
+                    return Json(new { success = false, result = "Failure", message = "Error receiving message." });
```
Comment (Horndev) [inline zapread.com/Controllers/MessagesController.cs]: `Response.StatusCode = (int)HttpStatusCode.BadRequest;`

**Comment by Horndev** [comment]
P.S.  I've never used github code review before - so I'm mostly just testing it with this patch.  Seems pretty slick from my end.

File: zapread.com/Controllers/MessagesController.cs (hunk 2 above)
Comment (wfinn) [reply zapread.com/Controllers/MessagesController.cs]: Edited, not tested yet as I'm not at home right now.

**Pull request merged**